src/controller.rs

```diff
@@ -171,9 +174,45 @@ impl AppState {
                 .into_iter()
                 .map(|u| Arc::new(RpcClient::new(get_http_url(u).expect("valid RPC url"))))
                 .collect(),
+            rpc_endpoint: endpoint.to_string(),
         }
     }
 
+    pub(crate) async fn sync_market_data_on_user_changes(&self, configured_markets: &[MarketId]) {
+        let default_sub_account = self.default_sub_account();
+        let state_commitment = self.tx_commitment;
+        let endpoint = self.rpc_endpoint.clone();
+        let configured_markets_vec = configured_markets.to_vec();
+        let self_clone = self.clone();
+        tokio::spawn(async move {
+            let pubsub_config = RpcAccountInfoConfig {
+                commitment: Some(state_commitment),
+                data_slice: None,
+                encoding: Some(UiAccountEncoding::JsonParsed),
+                min_context_slot: None,
+            };
+            let pubsub_client = PubsubClient::new(endpoint.replace("https", "wss").as_str())
+                .await
+                .expect("pubsub client init");
+            let (mut account_subscription, unsubscribe_fn) = pubsub_client
+                .account_subscribe(&default_sub_account, Some(pubsub_config))
+                .await
+                .expect("user account pubsub subscribe");
+
+            info!(target: LOG_TARGET, "Pubsub successfully subscribed to user account updates!");
+
+            // Process incoming account updates
+            while let Some(_) = account_subscription.next().await {
+                // refresh market subscriptions based on any new user positions
+                self_clone
+                    .subscribe_market_data(&configured_markets_vec)
+                    .await;
```

Comment:
can't remember if there's logs already but a debug log here to say something like 'updating subscriptions' would helpful

Reply:
there is already a log for when we call this function, do you want an additional log then? Like "Updating due to changed order/position state"